engines/http.c

```diff
@@ -122,12 +137,12 @@ static struct fio_option options[] = {
 		.group    = FIO_OPT_G_HTTP,
 	},
 	{
-		.name     = "http_s3",
-		.lname    = "S3 extensions",
-		.type     = FIO_OPT_BOOL,
-		.help     = "Whether to enable S3 specific headers",
-		.off1     = offsetof(struct http_options, s3),
-		.def	  = "0",
+		.name     = "http_mode",
```

Comment:
This really wants to be a posval option type, that's much cleaner than just accepting a string and then checking if it matches one of the valid types. With posval, that comes automatically and no extra checking is needed. Might want to consider for https as as well, it could be "off", "on", "nossl" for instance.